### PR TITLE
feat: add redis auth option

### PR DESCRIPTION
**Pull request by YeRuyi**
add redis auth opts

### CODE REVIEW

**Comment by YeRuyi** [comment]
@sshaplygin pls check.

File: redis_broadcast.go [hunk 1]
```diff
@@ -80,12 +80,17 @@ type allRoomResponse struct {
...
+		redisOpts = append(redisOpts, redis.DialPassword(opts.Password))
+	}
+
+	pub, err := redis.Dial(opts.Network, addr, redisOpts...)
 	if err != nil {
 		return nil, err
 	}
 
-	sub, err := redis.Dial(opts.Network, addr)
+	sub, err := redis.Dial(opts.Network, addr, redisOpts...)
```
Comment (sshaplygin) [inline redis_broadcast.go:93]: 👍  nice design

File: redis_broadcast.go [hunk 2]
```diff
@@ -80,12 +80,17 @@ type allRoomResponse struct {
 
 func newRedisBroadcast(nsp string, opts *RedisAdapterOptions) (*redisBroadcast, error) {
 	addr := opts.getAddr()
-	pub, err := redis.Dial(opts.Network, addr)
+	redisOpts := make([]redis.DialOption, 0)
```
Comment (sshaplygin) [inline redis_broadcast.go:83]: pls use next code: 

`redisOpts := make([]redis.DialOption, 0)` -> `var redisOpts []redis.DialOption`

File: server.go [hunk 3]
```diff
@@ -28,7 +28,12 @@ func NewServer(opts *engineio.Options) *Server {
 // Adapter sets redis broadcast adapter.
 func (s *Server) Adapter(opts *RedisAdapterOptions) (bool, error) {
 	opts = getOptions(opts)
-	conn, err := redis.Dial(opts.Network, opts.getAddr())
+	redisOpts := make([]redis.DialOption, 0)
```
Comment (sshaplygin) [inline server.go:31]: pls use next code:

`redisOpts := make([]redis.DialOption, 0)` -> `var redisOpts []redis.DialOption`

**Comment by YeRuyi** [comment]
pls check commit. thks.

**Comment by YeRuyi** [comment]
@sshaplygin thanks.

**Review by sshaplygin — APPROVED** [review]
LGTM

**Pull request merged**